src/components/AnonymousChat/child/ChatItemTargetText.tsx

```diff
@@ -1,100 +1,157 @@
-    return <View style={styles.mr8}>{avatar}</View>;
-  }, []);
+    if (!isShowUserInfo) return <View style={styles.avatar} />;
+    return <View style={styles.mrBuble}>{avatar}</View>;
+  }, [isShowUserInfo, avatar]);
+
+  const onSwipeToReply = React.useCallback(
+    (direction: 'left' | 'right') => {
+      if (direction === 'right') return;
+      if (swipeableRef.current) swipeableRef.current?.close();
+      setReplyPreview({
+        id: data?.id ?? data?.message?.id,
+        user: {username},
+        message,
+        message_type: messageType as MessageType,
+        updated_at: time,
+        chatType
+      });
+    },
+    [data, username, time, message, messageType, chatType]
+  );
 
   return (
-    <View style={styles.chatContainer}>
-      {renderAvatar()}
-      <View style={styles.textContainer}>
-        {!isContinuous && (
-          <View testID="chat-item-user-info" style={styles.chatTitleContainer}>
-            <Text style={styles.userText}>{username}</Text>
-            <View style={styles.dot} />
-            <Text style={styles.timeText}>{time}</Text>
-          </View>
-        )}
-        <Text testID="chat-item-message" style={styles.text}>
-          {message}
-        </Text>
+    <Swipeable
+      testID="swipeable"
+      ref={swipeableRef}
+      friction={3}
+      overshootFriction={2}
+      onSwipeableOpen={onSwipeToReply}
+      renderLeftActions={replyIcon}>
+      <View style={containerStyle(false, isReplyPrompt)}>
+        <Animated.View
+          style={[
+            styles.wrapper,
+            animatedBubbleStyle as ViewStyle,
+            animatedPulseStyle as ViewStyle
+          ]}>
+          {renderAvatar()}
+
+          <ContextMenu
+            previewBackgroundColor="transparent"
+            style={{flex: 1}}
+            actions={contextMenuActions}
+            onPress={(e) => onContextMenuPressed(e, data, chatType)}>
+            <View style={styles.radius8}>
+              <View style={textContainerStyle(false)}>
+                {isShowUserInfo && (
+                  <View testID="chat-item-user-info" style={styles.chatTitleContainer}>
+                    <Text style={[styles.userText, textStyle(false)]}>{username}</Text>
+                    <View style={dotStyle(false)} />
+                    <Text testID="timestamp" style={[styles.timeText, textStyle(false)]}>
+                      {calculateTime(time, true)}
+                    </Text>
+                  </View>
+                )}
+
+                <ChatReplyView
+                  type={chatType}
+                  messageType={messageType}
+                  replyData={data?.reply_data}
+                />
+
+                <Text testID="chat-item-message" style={messageStyle(false, messageType)}>
+                  {message}
+                </Text>
+              </View>
+            </View>
+          </ContextMenu>
+        </Animated.View>
       </View>
-    </View>
+    </Swipeable>
   );
 };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new implementation of `ChatItemTargetText` introduces several improvements such as gesture handling, context menus, and animations. However, there are a few areas that could be improved:

1. **Error Handling**: There is no error handling in the `onSwipeToReply` function (lines 89-103). If any of the operations fail (e.g., `setReplyPreview`), it might cause the application to crash or behave unexpectedly.

2. **Performance**: The `renderAvatar` function (lines 84-87) is re-rendered every time the component updates, even if `isShowUserInfo` and `avatar` props don't change. This can lead to unnecessary re-renders and affect performance.

3. **Code Readability**: The `contextMenuActions` array (lines 76-82) is conditionally populated with an additional action if `messageType !== MESSAGE_TYPE_DELETED`. It would be more readable to define this array in one place using a ternary operator or similar construct.

Here are some suggestions to address these issues:

```diff


<!-- This is an auto-generated comment by OSS CodeRabbit -->